staging/src/k8s.io/cloud-provider/sample/basic_main.go

```diff
@@ -0,0 +1,107 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// This file should be written by each cloud provider.
+// For an minimal working example, please refer to k8s.io/cloud-provider/sample/basic_main.go
+// For an advanced example, please refer to k8s.io/cloud-provider/sample/advanced_main.go
+// For more details, please refer to k8s.io/kubernetes/cmd/cloud-controller-manager/main.go
+
+package sample
+
+import (
+	"fmt"
+	"math/rand"
+	"os"
+	"time"
+
+	"github.com/spf13/pflag"
+	"k8s.io/cloud-provider"
+	"k8s.io/cloud-provider/app"
+	"k8s.io/cloud-provider/options"
+	"k8s.io/component-base/cli/flag"
+	"k8s.io/component-base/logs"
+	_ "k8s.io/component-base/metrics/prometheus/clientgo" // load all the prometheus client-go plugins
+	_ "k8s.io/component-base/metrics/prometheus/version"  // for version metric registration
+	"k8s.io/klog/v2"
```

Comment:
I think we need a comment here to indicate the cloud provider should import their cloud provider implementation?

Reply:
Also please add these sort of comments in the advanced sample.

Reply:
Can we add a comment/link in the version in K/K private pointing to these?

Reply:
Also a simple README.md file in this directory explaining for "new" cloud providers that they should copy the sample into their repo and then import the rest of the code. Also pointing to the examples of how our extensibility works.

Reply:
Added more comments. Will add the README.md file later as well. Thanks